csharp/src/ProtocolBuffers/CodedInputStream.cs

```diff
@@ -435,24 +520,16 @@ public long ReadSInt64()
             return DecodeZigZag64(ReadRawVarint64());
         }
 
-        private bool BeginArray(uint fieldTag, out bool isPacked, out int oldLimit)
+        /// <summary>
+        /// Reads a length for length-delimited data.
+        /// </summary>
+        /// <remarks>
+        /// This is internally just reading a varint, but this method exists
+        /// to make the calling code clearer.
+        /// </remarks>
+        public int ReadLength()

```

Comment:
I like having an extra method just for reading length, it really improves the readability!